src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -219,9 +219,18 @@ public String getBranch() {
     }
 
     public void setBranch(String branch) {
+        updateDisplayName(branch);
         this.branch = branch;
     }
 
+    /**
+     * Update display name if default current working branch is being used.
```

Comment:
implementation doesn't seems to be checking whether default branch is being used.